### PR TITLE
Move OME lower for non-transparent functions

### PR DESCRIPTION
**Pull request by meg-gupta**
Move it just before outliner which has not yet been migrated

### CODE REVIEW

**Comment by meg-gupta** [comment]
@swift-ci benchmark

**Comment by meg-gupta** [comment]
@swift-ci benchmark

File: lib/SILOptimizer/PassManager/PassPipeline.cpp [hunk 1]
```diff
@@ -522,6 +522,18 @@ static void addPerfEarlyModulePassPipeline(SILPassPipelinePlan &P) {
   // not blocked by any other passes' optimizations, so do it early.
   P.addDifferentiabilityWitnessDevirtualizer();
 
+  // Start by linking in referenced functions from other modules.
+  P.addPerformanceSILLinker();
+
```
Comment (meg-gupta) [inline lib/SILOptimizer/PassManager/PassPipeline.cpp:527]: TODO: Need to get to rid of deserialization hooks for OME

**Review by atrick — APPROVED** [review]
good progress

**Comment by meg-gupta** [comment]
@swift-ci test

**Comment by meg-gupta** [comment]
@swift-ci test source compatibility

**Comment by meg-gupta** [comment]
@swift-ci benchmark

**Comment by meg-gupta** [comment]
@swift-ci test

**Comment by meg-gupta** [comment]
@swift-ci Please Test Source Compatibility

**Pull request merged**